netmiko/extreme/extreme_ssh.py

```diff
@@ -1,13 +1,123 @@
 """Extreme support."""
 from __future__ import unicode_literals
+import re
+import time
 from netmiko.cisco_base_connection import CiscoSSHConnection
 
 
+QUESTIONS = {
+    'save': 'Do you want to save configuration to',
+}
+
+
 class ExtremeSSH(CiscoSSHConnection):
     """Extreme support."""
+
     def session_preparation(self):
-        """Extreme requires enable mode to disable paging."""
+        """
+        Extreme requires enable mode to disable paging.
+        """
         self._test_channel_read()
-        self.enable()
         self.set_base_prompt()
         self.disable_paging(command="disable clipaging\n")
+
+    def answer_questions(self, *args, **kwargs):
+        """
+        Send questions/answers with send_command_timing.
+        """
+        if len(args) >= 1:
+            command_string = args[0]
+
+        # handle commands which return a question
+        output = self.send_command_timing(command_string)
+        if QUESTIONS[command_string] in output:
+            output += self.send_command_timing('yes')
+
+        return output
+
+    def send_command(self, *args, **kwargs):
+        """
+        Catch questions from the CLI.
+        Handle prompt changes during command executions.
+        """
+        if len(args) >= 1:
+            command_string = args[0]
+
+        # command that prompt a question
+        if command_string in QUESTIONS.keys():
+            output = self.answer_questions(command_string)
+
+        # all other commands
+        else:
+            # Turn off 'auto_find_prompt' which loses the striped base prompt
+            kwargs.update({'auto_find_prompt': False})
+            # refresh prompt, which could have been changed on previous command
+            self.set_base_prompt()
+            output = super(ExtremeSSH, self).send_command(*args, **kwargs)
+
+        return output
+
+    def send_config_set(self, config_commands=None, exit_config_mode=True, delay_factor=1,
+                        max_loops=150, strip_prompt=False, strip_command=False):
+        """
+        Overwrite send_config_set to acomodate for commands which return a question
+        and expect a reply.
+        """
+        delay_factor = self.select_delay_factor(delay_factor)
+        if config_commands is None:
+            return ''
+        if not hasattr(config_commands, '__iter__'):
+            raise ValueError("Invalid argument passed into send_config_set")
+
+        # Send config commands
+        output = ''
+        for command_string in config_commands:
+
+            # command that prompt a question
+            if command_string in QUESTIONS.keys():
+                output += self.answer_questions(command_string)
+
```

Comment:
Same issue as above, Netmiko already has solutions for these, and this gets too hard for me to maintain for all of the various device types).